### PR TITLE
nixos manual: releases workflow improvements

### PR DESCRIPTION
**Pull request by vcunat**
- `git push --tags` is too dangerous (pushes all your tags)

### CODE REVIEW

**Comment by samueldr** [comment]
I would use `git checkout -b release-17.09` instead of `-B`. as following the advice in e.g. a sleep-deprived state of mind, could replace the branch locally! (Minor headache as it should be a protected branch.)

Let's keep this open for other gotchas and improvements.

**Comment by vcunat** [comment]
:+1:  I didn't even remember there was a `-B` variant of the command.  It seems I didn't read this properly half a year ago.

**Comment by samueldr** [comment]
* Update versionSuffix in nixos/release.nix, use git log --format=%an|wc -l to get the commit count

This seemingly isn't done anymore.

 * https://github.com/NixOS/nixpkgs/blob/release-18.03/nixos/modules/misc/version.nix

**Comment by samueldr** [comment]
```
[17:06:20] <gchristensen> ^ should expand the docs to include nixpkgs-N-darwin
```

**Comment by vcunat** [comment]
I did it on the master branch apparently :thinking:   https://github.com/NixOS/nixpkgs/commit/7e968a47cd4383919ac1fdf4b675024787506c8c#diff-831e8d9748240fb23e6734fdc2a6d16eR1

At the very least, the instructions should be clearer about which branch should take which of the changes.

**Comment by samueldr** [comment]
I made it like the linked 16.09 did:

 * https://github.com/NixOS/nixpkgs/pull/45958/files

**Comment by samueldr** [comment]
I'm thinking this should probably be scripted, at least the repository manipulation + local updates. (I'll think about it more and look into it later.)

**Comment by vcunat** [comment]
- [ ] update this https://github.com/NixOS/nixpkgs/pull/46192#issuecomment-419224640

**Comment by samueldr** [comment]
Added some notes; we're left with "what's up with `versionSuffix`", mainly.

**Comment by samueldr** [comment]
See #48222 → what about AMIs (and possibly other out-of-hydra artifacts?)

**Comment by vcunat** [comment]
Eh, so the instruction would be to ping Eelco about the AMIs? :-)  @edolstra: can we do better to offload that from you?  I don't know what's required to add them.

**Comment by samueldr** [comment]
We might want, in the next releases, to also link to the announcement on the discourse.

 * https://logs.nix.samueldr.com/nixos-dev/2018-11-11#1541894976-1541894809;

**Comment by zimbatm** [comment]
This pull request has been mentioned on **Nix community**. There might be relevant details there:

https://discourse.nixos.org/t/upload-recent-images-to-gs-nixos-images/1435/5

**Comment by zimbatm** [comment]
This pull request has been mentioned on **Nix community**. There might be relevant details there:

https://discourse.nixos.org/t/upload-recent-images-to-gs-nixos-images/1435/5

**Comment by zimbatm** [comment]
This pull request has been mentioned on **Nix community**. There might be relevant details there:

https://discourse.nixos.org/t/discourse-github-integration/1438/2

**Comment by vcunat** [comment]
- [ ] ping @rbvermaa about `gs://nixos-cloud-images`.  (Noone else who can do it is known to me ATM.)

**Comment by vcunat** [comment]
Oops, we should have finished this earlier.  /cc @lheckemann in case you don't know about this WIP.

**Review by lheckemann — APPROVED** [review]

**Comment by zimbatm** [comment]
can we merge this as it's already an improvement?

**Comment by vcunat** [comment]
Solved a simple conflict, and the diff seems to make sense to me.  @disassembler, @lheckemann: can you have a look if the changes match your "experience"?  /cc @asymmetric (hackday topic)

File: nixos/doc/manual/development/releases.xml [hunk 1]
```diff
@@ -169,8 +169,8 @@
     </listitem>
     <listitem>
      <para>
-      Change <literal>stableBranch</literal> to true and wait for channel to
-      update.
+      Change <literal>stableBranch</literal> to true in Hydra and wait for
```
Comment (asymmetric) [inline nixos/doc/manual/development/releases.xml]:
```suggestion
      Change <literal>stableBranch</literal> to <literal>true</literal> in Hydra and wait for
```

File: nixos/doc/manual/development/releases.xml [hunk 2]
```diff
@@ -169,8 +169,8 @@
...
+      channel to update.
```
Comment (asymmetric) [inline nixos/doc/manual/development/releases.xml]:
```suggestion
      the channel to update.
```

File: nixos/doc/manual/development/releases.xml [hunk 3]
```diff
@@ -212,7 +214,8 @@
     </listitem>
     <listitem>
      <para>
-      Send an email to nix-dev to announce the release with above information.
+      Create a new topic on <link xlink:href="https://discourse.nixos.org/">the
+      Discourse instance</link> to announce the release with above information.
```
Comment (asymmetric) [inline nixos/doc/manual/development/releases.xml]:
```suggestion
      Discourse instance</link> to announce the release with the above information.
```

**Pull request merged**